### PR TITLE
WIP: Jenkin files for builds across all projects in monorepo

### PR DESCRIPTION
**Pull request by axsaucedo**
This PR contains an extension to the previous work done with Jenkins, with the objective to automate the continuous integration, release and deployment process. This is the first iteration which has as an initial objective to include the Python seldon-core package so all tests and builds are ran automatically. This PR also provides a first iteration on how the model servers can be included in the build process. Below is a summary of what this PR includes, as well as next steps and caveats.

## New jobs:
image

## Python builder (two docker agents)
image

## What was done:
* Updated core-builder to add the s2i binaries (so it can be used to build model-servers)
* Updated Jenkins servers to latest (necessary to upgrade Pipeline plugins)
* Updated requirements.txt file to mirror latest setup.py and aligned tensorflow version (to latest)
* Removed unused parameters from seldon-core-engine Jenkinsfile

## Things to do when PR is landed:
* Change the build job branch to master for all jenkins jobs (currently set to this branch)
* Delete previous seldon-core build job(s)
* More of a heads up: The Java wrapper docker image is the only place where it wasn't upgrade to the newest 0.4 verison 

## Further work to be discussed:
* Pre-package model server build process in Jenkins 
* Building and pushing platform-specific images (redhat, etc)
* Build process for secondary images (core-builder, python-wrapper, etc) - as might be needed as platform-specific images
* Adding end-to-end tests into the process (ideally with k8s)
* Automating CI for new PRs (there now seems to be secure ways of connecting Jenkins a la Travis)
* Automatic deployment of latest to our k8s (demo) cluster (perhaps ArgoCD for that)
* Adding manual steps from Release process (tagging, changing helm chart, etc) to Jenkins pipeline
* Migrating Jenkins to our K8s cluster(s)
* Leveraging kubernetes in jobs
    * We can use k8s plugin instead of docker plugin (easy win to run all jobs in k8s)
    * Potentially moving some complexity into Argo jobs
* Build process for other projects outside mono-repo (Seldon Operator, Deploy, etc)

### CODE REVIEW

File: python/requirements.txt [hunk 1]
```diff
@@ -1,14 +1,17 @@
-numpy==1.14.5
```
Comment (cliveseldon) [inline python/requirements.txt]: Why have we removed versions

Comment (axsaucedo) [reply python/requirements.txt]: As discussed it was done to sync versions across setup.py and requirements.txt - I created #767 for this one

**Comment by axsaucedo** [comment]
Closing PR as we have gone forward with Jenkins X which has been activated with the repo

**Comment by seldondev** [comment]
failed to trigger Pull Request pipeline
* failed to create agent
* failed to calculate in repo config
* failed to load trigger config for repository SeldonIO/seldon-core for ref 43754f58e6da9cf2e2bbd0d8d94e7a29538de728
* failed to find any lighthouse configuration files in repo SeldonIO/seldon-core at sha 43754f58e6da9cf2e2bbd0d8d94e7a29538de728
* failed to process repo SeldonIO/seldon-core refref 43754f58e6da9cf2e2bbd0d8d94e7a29538de728
* failed to list files in directory /var/tmp/gitrepo283418320/.lighthouse
* open /var/tmp/gitrepo283418320/.lighthouse
* no such file or directory

**Pull request closed without merging**